java/pom.xml

```diff
@@ -1226,6 +1226,17 @@
       </build>
     </profile>
 
+    <profile>
+      <!-- Experimental JDK 21+ features -->
+      <id>jdk21+</id>
+      <activation>
+        <jdk>[21,]</jdk>
+      </activation>
+      <modules>
+        <module>memory/memory-ffm</module>
```

Comment:
Is this the best way to enable developers to build arrow-memory-ffm? I haven't come up with a better alternative, but I'm also not a POM expert.